gamification-evm-services/src/main/java/io/meeds/evm/gamification/service/EvmTriggerService.java

```diff
@@ -18,72 +18,63 @@
-import jakarta.annotation.PostConstruct;
 
-import jakarta.annotation.PreDestroy;
 import org.exoplatform.wallet.service.WalletAccountService;
 import org.exoplatform.wallet.model.Wallet;
 import io.meeds.gamification.model.EventDTO;
 import io.meeds.gamification.service.EventService;
 import org.apache.commons.collections.CollectionUtils;
 import org.apache.commons.lang3.StringUtils;
-import org.eclipse.jetty.util.thread.QueuedThreadPool;
 import org.exoplatform.commons.api.persistence.ExoTransactional;
 import org.exoplatform.services.listener.ListenerService;
 import org.exoplatform.services.log.ExoLogger;
 import org.exoplatform.services.log.Log;
 import org.exoplatform.social.core.identity.model.Identity;
 import org.exoplatform.social.core.manager.IdentityManager;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
+import org.springframework.stereotype.Service;
+import org.springframework.scheduling.concurrent.ThreadPoolTaskExecutor;
 
-@Component
+
+@Service
 public class EvmTriggerService {
 
-  private static final Log   LOG                        = ExoLogger.getLogger(EvmTriggerService.class);
+  private static final Log     LOG                        = ExoLogger.getLogger(EvmTriggerService.class);
 
-  public static final String GAMIFICATION_GENERIC_EVENT = "exo.gamification.generic.action";
+  public static final String   GAMIFICATION_GENERIC_EVENT = "exo.gamification.generic.action";
 
   @Autowired
-  private IdentityManager identityManager;
+  private IdentityManager      identityManager;
 
   @Autowired
-  private ListenerService listenerService;
+  private ListenerService      listenerService;
 
   @Autowired
-  private EventService eventService;
+  private EventService         eventService;
 
   @Autowired
-  private WalletAccountService walletAccountService;
-
-  private ExecutorService executorService;
+  private WalletAccountService     walletAccountService;
 
-  @PostConstruct
-  public void init() {
-    QueuedThreadPool threadFactory = new QueuedThreadPool(5, 1, 1);
-    threadFactory.setName("Gamification - Evm connector");
-    executorService = Executors.newCachedThreadPool(threadFactory);
-  }
+  @Autowired
+  private ThreadPoolTaskExecutor   threadPoolTaskExecutor;
 
-  @PreDestroy
-  public void stop() {
-    if (executorService != null) {
-      executorService.shutdownNow();
-    }
-  }
+  @Autowired
+  private ScheduledExecutorService scheduledExecutorService;
 
   /**
    * Handle evm trigger asynchronously
    *
    * @param evmTrigger evm retrieved trigger
    */
   public void handleTriggerAsync(EvmTrigger evmTrigger) {
-    executorService.execute(() -> handleTriggerAsyncInternal(evmTrigger));
+    scheduledExecutorService.schedule(() -> threadPoolTaskExecutor.execute(() -> handleTriggerAsyncInternal(evmTrigger)),
+                                      1,
+                                      TimeUnit.SECONDS);
```

Comment:
Why this change, can you explain please why differing execution ?

Reply:
I had added this step to force the synchronization of the various actions (I had a synchronization problem when I had multiple actions, this problem only occurs when I use hsql db and not mysql).

Reply:
ok, let's discuss more about it.